cmd/influxd/run/server.go

```diff
@@ -508,6 +518,43 @@ func (s *Server) reportServer() {
 	go client.Post("http://m.influxdb.com:8086/db/reporting/series?u=reporter&p=influxdb", "application/json", data)
 }
 
+// registerServer registers the server on start-up.
+func (s *Server) registerServer() error {
+	if s.enterpriseToken == "" {
+		return nil
+	}
+
+	clusterID, err := s.MetaStore.ClusterID()
+	if err != nil {
+		log.Printf("failed to retrieve cluster ID for registration: %s", err.Error())
+		return err
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+
+	j := map[string]interface{}{
+		"cluster_id": clusterID,
+		"server_id":  s.MetaStore.NodeID(),
+		"host":       hostname,
+		"product":    "influxdb",
+		"version":    s.buildInfo.Version,
+	}
+	b, err := json.Marshal(j)
+	if err != nil {
+		return err
+	}
+
+	url := fmt.Sprintf("%s/api/v1/servers?token=%s", s.enterpriseURL, s.enterpriseToken)
+	client := http.Client{Timeout: time.Duration(5 * time.Second)}
+	if _, err := client.Post(url, "application/json", bytes.NewBuffer(b)); err != nil {
```

Comment:
This won't check for errors coming from Enterprise. In Go the `error` returned from `Post` is defined as the following:

```
An error is returned if caused by client policy (such as CheckRedirect), or if there was an HTTP protocol error. A non-2xx response doesn't cause an error.
```

You'll probably want to look at the status code and then handle the body of a non 2xx. 

Here's what errors look like from Enterprise: https://github.com/influxdb/enterprise#api-errors